### PR TITLE
localize reward message (You received ~)

### PR DESCRIPTION
**Pull request by EnochG1**
## What are the changes?
Add the option to localize message text when you receive modifier.

## Why am I doing these changes?
All text should be localizable.

## What did change?
 - Change the code for localize.
 - Added localization key in each language folders.
 - Translated it to Korean.

### Screenshots/Videos

| Lang | Result |
| :-----: | :---: |
| En | image |
| De | image |
| Es | image |
| Fr | image |
| Ko | image |
| pt_BR | image |
| zh_CN | image |

## How to test the changes?
Manually.
Try wave clear with a fixed reward, like
classic 8 : rival (EXP. All)
classic 20 : Gym leader (Super EXP. Charm)
(find onReward of ModifierRewardPhase)

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/zh_CN/battle.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultOneHitKO": "一击必杀！",
   "attackFailed": "但是失败了！",
   "attackHitsCount": "击中 {{count}} 次！",
+  "rewardGain": "You received\n{{modifierName}}!",
```
Comment (40chyan) [inline src/locales/zh_CN/battle.ts:28]:
```suggestion
  "rewardGain": "你获得了\n{{modifierName}}！",
```

Comment (40chyan) [reply src/locales/zh_CN/battle.ts:28]: cn translation

File: src/locales/es/battle.ts [hunk 2]
```diff
@@ -25,6 +25,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultOneHitKO": "¡KO en 1 golpe!",
   "attackFailed": "¡Pero ha fallado!",
   "attackHitsCount": "N.º de golpes: {{count}}.",
+  "rewardGain": "You received\n{{modifierName}}!",
```
Comment (GoldTra) [inline src/locales/es/battle.ts:28]:
```suggestion
  "rewardGain": "¡Has obtenido\n{{modifierName}}!",
```

**Review by GoldTra — CHANGES_REQUESTED** [review]
Spanish translation

**Review by GoldTra — APPROVED** [review]

File: src/locales/de/battle.ts [hunk 3]
```diff
@@ -25,6 +25,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultOneHitKO": "Ein K.O.-Treffer!",
   "attackFailed": "Es ist fehlgeschlagen!",
   "attackHitsCount": "{{count}}-mal getroffen!",
+  "rewardGain": "You received\n{{modifierName}}!",
```
Comment (CodeTappert) [inline src/locales/de/battle.ts:28]:
```suggestion
  "rewardGain": "Du erhältst\n{{modifierName}}!",
```

File: src/locales/fr/battle.ts [hunk 4]
```diff
@@ -25,6 +25,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultOneHitKO": "K.O. en un coup !",
   "attackFailed": "Mais cela échoue !",
   "attackHitsCount": "Touché {{count}} fois !",
+  "rewardGain": "You received\n{{modifierName}}!",
```
Comment (Adri1) [inline src/locales/fr/battle.ts:28]:
```suggestion
  "rewardGain": "Vous recevez\n{{modifierName}} !",
```

**Review by Adri1 — COMMENTED** [review]
Added French

File: src/locales/pt_BR/battle.ts [hunk 5]
```diff
@@ -25,6 +25,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultOneHitKO": "Foi um nocaute de um golpe!",
   "attackFailed": "Mas falhou!",
   "attackHitsCount": "Acertou {{count}} vezes.",
+  "rewardGain": "You received\n{{modifierName}}!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/battle.ts:28]: Suggested portuguese translation
```suggestion
  "rewardGain": "Você recebeu\n{{modifierName}}!",
```

**Review by Adri1 — APPROVED** [review]

**Comment by EnochG1** [comment]
@CodeTappert
@40chyan
@josericardo-fo

Could you approve this request if updated screenshot of your language at main article is correct?

**Review by CodeTappert — APPROVED** [review]

**Review by josericardo-fo — APPROVED** [review]

**Review by 40chyan — APPROVED** [review]
zh_cn approved

**Review by Tempo-anon — APPROVED** [review]

**Pull request merged**